### PR TITLE
[stable7] Close cursor when setting preference value

### PR DESCRIPTION
**Pull request by PVince81**
Similar fix like https://github.com/owncloud/core/pull/13772 but for stable7.

I found this while investigating https://github.com/owncloud/core/issues/12801#issuecomment-72246486

@MorrisJobke @icewind1991 @Deepdiver1975 @LukasReschke

### CODE REVIEW

**Comment by DeepDiver1975** [comment]
@PVince81 let's postpone backports regarding cursor closes for the time being - we first need to get the reasoning behind oracle failing properly sorted out (most probably related to different doctrine/dbal versions we use in stable7 and master).

This is something we might want to look into after OC8 has been released - THX

**Comment by PVince81** [comment]
@DeepDiver1975 I'm doing this to hopefully fix @fleish's update issues from OC6->OC7 regarding LDAP which might or might not break itself due to such database is locked issues. We might not want to merge this right now but at least he can try the patch. See https://github.com/owncloud/core/issues/12801#issuecomment-72421311

**Comment by PVince81** [comment]
Still need to have a look at the Oracle issues.

**Comment by DeepDiver1975** [comment]
> Still need to have a look at the Oracle issues.

a work around might be to only close the cursor for non-oracle dbs

**Comment by MorrisJobke** [comment]
@owncloud-bot retest this please

File: lib/private/preferences.php [hunk 1]
```diff
@@ -173,10 +173,16 @@ public function getValue($user, $app, $key, $default = null) {
...
-		$exists = $oldValue !== false;
+		$result = $this->conn->executeQuery($query, array($user, $app, $key));
+		$row = $result->fetch();
+		$result->closeCursor();
+		if ($row) {
+			$oldValue = $row['configvalue'];
+			$exists = true;
+		}
 
 		if($oldValue === strval($value)) {
```
Comment (MorrisJobke) [inline lib/private/preferences.php]: `$oldValue` doesn't exists here if `$row` is empty.

**Comment by PVince81** [comment]
Upgrading to 8.0.x is advised.
Closing.

**Pull request closed without merging**